packages/zoe/src/zoeService/zoe.js

```diff
@@ -146,27 +160,66 @@ function makeZoe(vatAdminSvc) {
           );
           return invitationMint.mintPayment(invitationAmount);
         },
-        // checks of keyword done on zcf side
+        // checks if keyword done on zcf side
         saveIssuer: (issuerP, keyword) =>
           issuerTable
             .getPromiseForIssuerRecord(issuerP)
             .then(({ issuer, brand }) => {
-              instanceRecord.issuerKeywordRecord = {
-                ...instanceRecord.issuerKeywordRecord,
-                [keyword]: issuer,
-              };
-              instanceRecord.brandKeywordRecord = {
-                ...instanceRecord.brandKeywordRecord,
-                [keyword]: brand,
-              };
-              if (!brandToPurse.has(brand)) {
-                brandToPurse.init(brand, E(issuer).makeEmptyPurse());
-              }
+              registerIssuerStuff(keyword, issuer, brand);
             }),
         shutdown: () => {
           exitAllSeats();
           adminNode.terminate();
         },
+        makeZoeMint: (keyword, mathHelperName = 'nat') => {
+          assert(
+            !(keyword in instanceRecord.issuerKeywordRecord),
+            details`Keyword ${keyword} already registered`,
+          );
+
+          // Local indicates the one zoe itself makes from vetted code,
+          // and so can be assumed correct and fresh by zoe.
+          const {
+            mint: localMint,
+            issuer: localIssuer,
+            amountMath: localAmountMath,
+            brand: localBrand,
+          } = makeIssuerKit(keyword, mathHelperName);
+          const localIssuerRecord = harden({
+            brand: localBrand,
+            issuer: localIssuer,
+            amountMath: localAmountMath,
+          });
+          issuerTable.registerIssuerRecord(localIssuerRecord);
+          registerIssuerStuff(keyword, localIssuer, localBrand);
+          const zoeMint = harden({
+            getIssuerRecord: () => {
+              return localIssuerRecord;
+            },
+            mintAllocation: (allocation, zoeSeat = undefined) => {
+              // Malformed amount (or anything that might cause mintPayment
+              // to fail later) must fail now instead.
+              amount = localAmountMath.coerce(amount);
+              const pooledPurseP = brandToPurse.get(localBrand);
+              const payment = localMint.mintPayment(amount);
+              E(pooledPurseP)
+                .deposit(payment, amount)
+                .catch(reason => {
+                  // TODO panic?
+                  // If this delayed local deposit fails, that's very bad news.
+                  throw reason;
+                });
+              // Proceed assuming the deposit succeeded
+              const oldAmount = undefined; // where from?
```

Comment:
I think you have to ask the pooledPurse for its balance before or after the deposit above.  

Since the Purse and Issuer are known to be local, is there a way to get a prompt response from them? Are we forced to use `.then()` because the purse was stored using `brandToPurse.init(brand, E(issuer).makeEmptyPurse());`?

The alternative would be to store zoe-based mints separately from remote mints so they can be accessed synchronously. Or to cache the balance, since we know that it can only be changed by calls through this purse.

Reply:
Thanks @Chris-Hibbert I refactored this bit so that we have immediate access to that pooled purse.